lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -432,47 +432,41 @@ def operations
       end
 
       define_method("find_all_of_type_#{pe_type}") do |options = {}|
+        options[:unique_identifier] = options.delete(:uuid) if options[:uuid]
         conditions = options.slice!(:per_page, :page, :ignore_case).stringify_keys
+        # Separate conditions on PolicyElement columns from conditions on the extra_attributes column
         extra_attribute_conditions = conditions.slice!(*PolicyElement.column_names)
         include_conditions, conditions = conditions.partition { |k,v| include_condition?(k,v) }
+        # Generated PolicyElement class
         pe_class = class_for_type(pe_type)
 
-        # Arel matches provides agnostic case insensitive sql for mysql and postgres
-        all = if options[:ignore_case]
-                conditions.map do |k,v|
-                  if ignore_case_applies?(options[:ignore_case], k)
-                    pe_class.arel_table[k].matches(v)
-                  else
-                    pe_class.arel_table[k].eq(v)
-                  end
-                end.reduce(pe_class.where(nil)) { |rel, e| rel.where(e) }
-              else
-                pe_class.where(conditions.to_h)
-              end
-
-        include_conditions.each do |key, value|
-          all = Adapter.apply_include_condition(scope: all, key: key, value: value[:include], klass: class_for_type(pe_type))
-        end
+        results = build_active_record_relation(
```

Comment:
`results` seems pretty ambiguous. @hstebbins-mdsol what do you think about `active_record_relation` instead?

Reply:
excellent idea